packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -348,10 +346,9 @@ const App = (props) => {
 
                         <Box id="app" display="flex" flexDirection="column" flex={1}>
                             <SkipNavLink zIndex="skipLink">Skip to Content</SkipNavLink>
-                            <StoreLocatorModal
-                                isOpen={isOpenStoreLocator}
-                                onClose={onCloseStoreLocator}
-                            />
+                            {isOpenStoreLocator && (
+                                <StoreLocatorModal onClose={onCloseStoreLocator} />
+                            )}
```

Comment:
I'm actually surprised the Chakra Modal renders child component when `isOpen` from the `useDisclosure` hook is false?

Is this the reason you are making this change?

Reply:
It's an established pattern to have both `isOpen` and `onClose` as a pair of props to manage the modal. It's kinda weird to only have `onClose` and have a `<Modal open={true}>` hard coded `true` in the implementation.

Have you considered keeping the previous api:

```
<StoreLocatorModal isOpen={isOpenStoreLocator} onClose={onCloseStoreLocator} />
```

and in the implementation you can control the rendering of the modal and its content like:
```
const StoreLocatorModal = ({isOpen}) => {
   return {isOpen ? <div>...</div> : null}
}
```

Reply:
Yeah in the last PR this pattern was followed, but with this pattern the hook calls that occur at render of the modal occurs at site visit, even when the modal is not opened. The concern on fetching data needlessly on site visit was brought up in this [Slack thread](https://salesforce-internal.slack.com/archives/C06V992DR1N/p1719008830241409?thread_ts=1719007850.268619&cid=C06V992DR1N), and the solution to that mentioned in the thread and implemented here is to conditionally load the modal when the store locator button is clicked, which means the hooks will only be called when the modal is physically open.

Reply:
You can conditionally render based on the `isOpen` flag:

```js
const StoreLocatorModal = ({isOpen}) => {
   return {isOpen ? <div>...</div> : null}
}
```

Reply:
Ohh yeah adding another component layer would work. Thank you.

Reply:
i don't think you need another layer, you can do it directly in `StoreLocatorModal`

Reply:
I made the changes locally and am still seeing that on site visit the hook requests are still being made. The hook calls are in the `StoreLocatorModal` index.jsx file, so I'll probably have to make a layer and have `<YourModalThatFetchesThruHooks/>` be the `StoreLocatorModal` file 🤔

Reply:
Something else must be wrong...

see a minimal example: https://playcode.io/1920443

Reply:
Ah i see. I thought the data fetching happens in the `StoreLocatorContent` component

Reply:
Thoughts for moving the data fetching hooks to `StoreLocatorContent` component?

 it does look like there are a bunch of stuff that is passed down to `StoreLocatorContent`, this way you don't need that middle layer `StoreLocatorModalWithHooks`...

Reply:
This is what I mean...

# Before

```jsx
const StoreLocatorModal = ({isOpen}) => {
   return {isOpen ? <StoreLocatorModalWithHooks/> : null}
}

const StoreLocatorModalWithHooks = () => {
    const data = useSearchStore()
    return <StoreLocatorContent data={data}>
}

const StoreLocatorContent = (data) => {
    return ...
}
```

# after
```
const StoreLocatorModal = ({isOpen}) => {
   return {isOpen ? <StoreLocatorContent/> : null}
}

const StoreLocatorContent = () => {
    const data = useSearchStore()
    return ...
}
```